instrumentation-core/servlet-2.2/src/test/java/io/opentelemetry/instrumentation/servlet/ServletHttpServerTracerTest.java

```diff
@@ -0,0 +1,64 @@
+package io.opentelemetry.instrumentation.servlet;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.Test;
+
+public class ServletHttpServerTracerTest {
+
+  @Test
+  void testGetSpanName_emptySpanName() {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    when(request.getServletPath()).thenReturn("");
+    when(request.getMethod()).thenReturn("PUT");
+    String spanName = ServletHttpServerTracer.getSpanName(request);
+    assertThat(spanName).isEqualTo("HTTP PUT");
+  }
+
+  @Test
+  void testGetSpanName_nullSpanName() {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    when(request.getServletPath()).thenReturn(null);
+    assertThatThrownBy(() -> ServletHttpServerTracer.getSpanName(request)).isInstanceOf(NullPointerException.class);
+  }
+
```

Comment:
not sure if this behavior is worth codifying in a unit test?

Reply:
Yeah, I know, it does seem kinda silly...but I was speculating about a code path and also wanted to make sure I didn't change this.  I dunno...I am happy to remove it if it's annoying and you hate it.   😛